### DISCUSSION TITLE
Redirection local time in RDP session in freerdp2-x11 2.11.2 doesn't work

### DISCUSSION THREAD

**Discussion by fleurmari** [FreeRDP/FreeRDP#9933] [Q&A]
I can't redirect local time in the session.
XfreeRDP log:
```
xfreerdp /v:*** /u:*** /p:*** /log-level:trace | grep -i time

[DEBUG][com.winpr.timezone] - tz: Bias=0 sn='Coordinated Universal Time' dln='Coordinated Universal Time'
```

And time in session like time on the server, not like on the client

**Comment by akallabeth** [comment]
looks like automatich matching of your unix timezone to a windows timezone fails?
* you can set it manually with the environment variable `TZ="Central Europe Standard Time"` to set it to `CET` (or any other windows timezone name)
* you can update `FreeRDP` and check if the mapping was updated
* you can check why your unix timezone (which one is it?) is not mapped to the windows timezones in `const TIME_ZONE_ENTRY TimeZoneTable`

**Reply by fleurmari** [reply]
Yes, it looks like a fail while matching a Unix timezone to a Windows timezone.

I tried the first way and it hasn't worked.
And then, I updated my linux (debian 12) by ```sudo apt update | sudo apt upgrade```.

freerdp2-x11 still has the same version.

And now, xfreerdp says:
```
[ERROR][com.winpr.timezone] - Unable to get current timezone rule
[DEBUG][com.winpr.timezone] - tz: Bias=-300 sn='Russia TZ 4 Standard Time' dln='Russia TZ 4 Daglight Time'
```

And time in session still has the server's timezone, not the client's.

And I don't know how to check why the unix timezone is not mapped to the windows timezone

**Reply by akallabeth** [reply]
for the mapping:
* what is in your `/etc/timezone`?

as for why the server time is different, with the messages above the client sends the preference to the server, but the server might be configured to ignore that. does it work with any other client to that server to get the correct time settings?

**Reply by fleurmari** [reply]
Yes, it works correctly with other clients (remmina, freerdp (2.3)).
My timezone is 'Asia/Yekaterinburg' (/etc/localtime -> /usr/share/zoneinfo/Asia/Yekaterinburg)

**Reply by akallabeth** [reply]
ok, tested this now on my fedora 40:
* `xfreerdp` from repo 2.11.5, 2.10.0 like in debian 12, 2.11.2 like you mentioned and current `master`
* `/etc/localtime -> /usr/share/zoneinfo/Asia/Yekaterinburg`

works as expected. (I do get the warning, but it does only mean the current daylight/standard bias can not be reliably determined)

please check that your windows is configured to use the timezone value of a RDP client:
`gpedit.msc` policy `Administrative Templates\Windows Components\Remote Desktop Services\Remote Desktop Session Host\Device and Resource Redirection\Allow time zone redirection` must be set to `Enabled`
